google_oauth_block.py

```diff
@@ -63,6 +75,15 @@ def _authenticate(self):
             self._access_token = self.get_access_token(self.get_google_scope())
             self._logger.debug("Obtained access token {0}".format(
                 self._access_token))
+
+            if self._reauth_job:
+                self._reauth_job.cancel()
+
+            # Remember to reauthenticate at a certain point if it's configured
+            if self.reauth_interval.total_seconds() > 0:
+                self._reauth_job = Job(
+                    self._authenticate, self.reauth_interval, False)
```

Comment:
Not based on the way it is set up. It's not that we want to re-authenticate every hour. It's that we want to re-authenticate one hour after the authentication succeeds. Normally these will be the same, but if the authentication fails and starts some type of retry pattern, they could fall out of sync. That's why I only made it a one-time-use job